macros/ui/niceTables.pl

```diff
@@ -3,835 +3,1201 @@ =head1 niceTables.pl
+	return "$left$separator$center$separator$right";
+}
 
-Usage:  LayoutTable(...)
-	See usage for DataTable. The HTML output will use section and div boxes instead of HTML tabling elements
-	Anything having to do with headers, captions, and data cells no longer make sense (although 'data' is still
-	used as the key for cell contents).
+sub prefix {
+	my ($center, $left, $separator) = @_;
+	$separator = "\n" unless ($separator);
+	return join("$separator", ($left, $center)) if ($left ne '');
+	return $center;
+}
 
-=cut
+sub suffix {
+	my ($center, $right, $separator) = @_;
+	$separator = "\n" unless ($separator);
+	return join("$separator", ($center, $right)) if ($right ne '');
+	return $center;
+}
 
-sub LayoutTable {
-	my $dataref = shift;
-	if   ($main::displayMode eq 'PTX') { DataTable($dataref, @_); }
-	else                               { DataTable($dataref, LaYoUt => 1, @_); }
+sub css {
+	my ($property, $value) = @_;
+	return ($value) ? "$property:$value;" : '';
 }
 
-sub TeX_Alignment_to_CSS {
-	my $alignmentstring = shift;
-	my $bracesregex     = qr/(\{(?>[^{}]|(?R))*\})/x;
-	# grabs outer level braces and their contents, including inner brace pairs
-	my $bracecontentsregex = qr/((?>[^{}]|(??{$bracesregex}))*)/x;
-	# grabs contents of an outer level brace pair, including inner brace pairs
-
-	my $css = '';
-	if ($alignmentstring =~ /r\s*/) {
-		$css .= "text-align:right; white-space:nowrap; ";
-	} elsif ($alignmentstring =~ /c\s*/) {
-		$css .= "text-align:center; white-space:nowrap; ";
-	} elsif ($alignmentstring =~ /l\s*/) {
-		$css .= "text-align:left; white-space:nowrap; ";
-	} elsif ($alignmentstring =~ /X\s*/) {
-		$css .= "text-align:justify; white-space:normal; ";
-	} elsif ($alignmentstring =~ /p\s*\{((??{$bracecontentsregex}))\}\s*/) {
-		$css .= "text-align:justify; white-space:normal; width:" . $1 . "; ";
+sub tag {
+	my ($inner, $name, $attributes, $separator) = @_;
+	$separator = "\n" unless defined $separator;
+	my $return = "<$name";
+	for my $x (main::lex_sort(keys %$attributes)) {
+		$return .= qq( $x="$attributes->{$x}") if ($attributes->{$x} ne '');
+	}
+	if ($inner) {
+		$return .= ">$separator";
+		$return .= $inner;
+		$return .= "$separator</$name>";
+	} else {
+		$return .= '>' unless ($main::displayMode eq 'PTX');
+		$return .= '/>' if ($main::displayMode eq 'PTX');
 	}
-	return $css;
+	return $return;
+}
+
+sub getRuleCSS {
+	my $input  = shift;
+	my $output = '';
+	if ($input =~ /^\s*(\.\d+|\d+\.?\d*)\s*$/) {
+		$output = "solid $1px" if $1;
+	} elsif ($input) {
+		$output = "solid $input" if $input;
+	}
+	return $output;
+}
+
+sub getPTXthickness {
+	my $input  = shift;
+	my $output = '';
+	if ($input == 1) {
+		$output = "minor";
+	} elsif ($input == 2) {
+		$output = "medium";
+	} elsif ($input >= 3) {
+		$output = "major";
+	} elsif ($input eq '0.04em') {
+		$output = 'minor';
+	} elsif ($input eq '0.07em') {
+		$output = 'medium';
+	} elsif ($input eq '0.11em') {
+		$output = 'major';
```

Comment:
In all of these conditionals `$input` can be undefined, and if so gives warnings about the use of an uninitialized value in string or numeric comparison.